src/Xamarin.Android.Tools.Aidl/CSharpCodeGenerator.cs

```diff
@@ -289,8 +303,8 @@ protected override bool OnTransact (int code, global::Android.OS.Parcel data, gl
 					w.WriteLine ("\t\t\t\t{0}", GetWriteStatements (method.ReturnType, "reply", "result", "global::Android.OS.ParcelableWriteFlags.ReturnValue"));
 				for (int i = 0; method.Arguments != null && i < method.Arguments.Length; i++) {
 					var a = method.Arguments [i];
-					if (a.Modifier == null || a.Modifier.Contains ("out"))
```

Comment:
I assume this change/removal is why `data.WriteLong (arg0);` is removed in the unit test diff.

I *think* this is correct?

Reply:
This change was necessary because before for primitive input parameters like in `void Authenticate (long operationId)` the input parameter after function execution was written back to the `reply` Parcel which is only reqired for out and inout parameters.